### PR TITLE
docs: add Envoy Gateway threat model

### PR DESCRIPTION
**Pull request by edurra**
**What this PR does / why we need it**:

This PR includes the Threat Model and End User Recommendations conducted by ControlPlane. The results of the Threat Modelling exercise have been included in the `thread-model.md` file.

**Which issue(s) this PR fixes**:
N/A

### CODE REVIEW

**Comment by arkodg** [comment]
thanks @edurra, as @Xunzhuo mentioned, lets move this under `site` so it automatically shows up on the website

recommend moving it to https://github.com/envoyproxy/gateway/tree/main/site/content/en/latest/user/security and another copy in https://github.com/envoyproxy/gateway/tree/main/site/content/en/v1.0.0/user/security

File: examples/threat-model/threat-model.md [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+
+### AT5
+
+![AT5](./images/AT5.png)
+
+### AT6
+
+![AT6](./images/AT6.png)
+
+## About
```
Comment (arkodg) [inline examples/threat-model/threat-model.md:651]: prefer if we moved `About`, `Threat Modeling Team` and `Reviewers` to the top to increase ethos
lets flatten out the names into a single line, to fit more of the introduction on the page

Comment (edurra) [reply examples/threat-model/threat-model.md]: Sure, I just did the modification.

**Comment by edurra** [comment]
Hi @Xunzhuo @arkodg , I just moved the content to the site directories as mentioned :+1: Let me know if any modification else is required

**Review by zirain — APPROVED** [review]

**Pull request merged**